bcipy/gui/mode/RSVPKeyboard.py

```diff
@@ -1,210 +1,281 @@
+            username = self.user_input.currentText()
+            experiment_type = self.sender().get_id()
             mode = 'RSVP'
             cmd = 'python bci_main.py -m {} -t {} -u {} -p {}'.format(
                 mode, experiment_type, username, self.parameter_location)
 
             subprocess.Popen(cmd, shell=True)
 
     def check_input(self) -> bool:
-        """Check Input."""
+        """Check Input.
+
+        Checks to make sure user has input all required fields. Currently, only user id is required.
+        """
         try:
-            if self.comboboxes[0].GetValue() == '':
-                dialog = wx.MessageDialog(
-                    self, "Please Input User ID", 'Info',
-                    wx.OK | wx.ICON_WARNING)
-                dialog.ShowModal()
-                dialog.Destroy()
+            if self.user_input.currentText() == RSVPKeyboard.default_text:
+                self.throw_alert_message(
+                    title='BciPy Alert',
+                    message='Please input a User ID',
+                    message_type=AlertMessageType.INFO,
+                    okay_to_exit=True)
                 return False
             if self.event_started:
                 return False
         except Exception as e:
-            dialog = wx.MessageDialog(
-                self, f'Error, {e}',
-                'Info', wx.OK | wx.ICON_WARNING)
-            dialog.ShowModal()
-            dialog.Destroy()
+            self.throw_alert_message(
+                title='BciPy Alert',
+                message=f'Error, {e}',
+                message_type=AlertMessageType.CRIT,
+                okay_to_exit=True)
             return False
         return True
 
-    def offline_analysis(self, _event: wx.Event) -> None:
+    def offline_analysis(self) -> None:
         """Run the offline analysis in a new Process."""
         cmd = 'python bcipy/signal/model/offline_analysis.py'
         self.event_started = True
         subprocess.Popen(cmd, shell=True)
 
-    def refresh(self, _event: wx.Event) -> None:
+    def refresh(self) -> None:
+        """Refresh.
+
+        Reset event_started to False to allow a new task to be selected.
+        """
         self.event_started = False
 
-    def load_items_from_txt(self, _event):
-        """Loads user directory names from the data path defined in
-        parameters.json, and adds those directory names as items to the user id
-        selection combobox."""
-        parameters = load_json_parameters(
-            self.parameter_location, value_cast=True)
-        data_save_loc = parameters['data_save_loc']
+    def load_items_from_txt(self) -> List[str]:
+        """Load Items From Text.
+
+        Loads user directory names below experiments from the data path defined in parameters.json
+        and returns them as a list.
+        """
+        saved_users = []
+        data_save_loc = self.parameters['data_save_loc']
+
```

Comment:
I just realized that the parameter_location gets updated, but the parameters never get reloaded, so this will always be the data_save_loc from the default parameters. I can fix that after this is merged.

Reply:
I added a bug ticket for this! https://www.pivotaltracker.com/story/show/175541919